db/collection.go

```diff
@@ -671,6 +666,153 @@ func (db *db) validateCollectionChanges(
 	return nil
 }
 
+var newCollectionValidators = []func(
+	client.CollectionDefinition,
+	map[string]client.CollectionDefinition,
+) error{
+	validateSecondaryFieldsPairUp,
+	validateRelationPointsToValidKind,
+	validateSingleSidePrimary,
+}
+
+func (db *db) validateNewCollection(
```

Comment:
:) Cheers

> thought: may be worth moving to db_validators.go at some point

I want to move them somewhere, and there are other improvements that can be made too, but I started doing it in here before pausing and deciding that it would require too much thought for an already fiddly PR (noted in the description, I'll be creating a ticket once people are happy with this PR being merged).